unit/unittype.go

```diff
@@ -322,25 +326,40 @@ func (u *Unit) Format(fs fmt.State, c rune) {
 	case 'e', 'E', 'f', 'F', 'g', 'G':
 		p, pOk := fs.Precision()
 		w, wOk := fs.Width()
-		if u.formatted == "" && len(u.dimensions) > 0 {
-			u.formatted = u.dimensions.String()
+		formatted := u.formattedUnits()
+		if formatted == "" && len(u.dimensions) > 0 {
+			formatted = u.dimensions.String()
```

Comment:
Note also that we still aren't concurrency safe because `NewDimension` can be called during formatting. See my comment in #902

Reply:
I missed that. You are right. This complicates things significantly.

Reply:
I have fixed this. It requires locking `symbols` and `dimensions` and only accessing those through a small set of functions.